fix: Resolve mainnet v19 fork issues

UdjinM6 opened the pull request:
## Issue being fixed or feature implemented

## What was done?
pls see individual commits

## How Has This Been Tested?
reorg mainnet around forkpoint with a patched client (to allow low difficulty), run tests

## Breaking Changes
Another evodb migration is required. Going back to an older version or migrating after the fork requires reindexing.

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [ ] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

ogabrielides commented on src/evo/deterministicmns.h:
```diff
@@ -428,14 +442,24 @@ class CDeterministicMNList
+            CBLSPublicKeyVersionWrapper wrapped_pub_key(local_pub_key, specific_legacy_bls_scheme.value());
+            oldHash = ::SerializeHash(wrapped_pub_key);
+        } else {
+            assert(!specific_legacy_bls_scheme.has_value());
```
This must be removed or changed to:
```suggestion
if (specific_legacy_bls_scheme.has_value()) assert(!specific_legacy_bls_scheme.value();
```

ogabrielides commented on src/evo/deterministicmns.h:
```diff
@@ -408,14 +412,24 @@ class CDeterministicMNList
+            CBLSPublicKeyVersionWrapper wrapped_pub_key(local_pub_key, specific_legacy_bls_scheme.value());
+            hash = ::SerializeHash(wrapped_pub_key);
+        } else {
+            assert(!specific_legacy_bls_scheme.has_value());
```
This must be removed or changed to:
```suggestion
if (specific_legacy_bls_scheme.has_value()) assert(!specific_legacy_bls_scheme.value();
```

ogabrielides commented on src/evo/deterministicmns.cpp:
```diff
@@ -540,10 +498,10 @@ void CDeterministicMNList::UpdateMN(const CDeterministicMN& oldDmn, const std::s
         throw(std::runtime_error(strprintf("%s: Can't update a masternode %s with a duplicate pubKeyOperator=%s", __func__,
-                oldDmn.proTxHash.ToString(), pdmnState->pubKeyOperator.Get().ToString())));
+                oldDmn.proTxHash.ToString(), pdmnState->pubKeyOperator.Get().ToString(pdmnState->nVersion == CProRegTx::LEGACY_BLS_VERSION))));
     }
```
Since this can be false in the future.

ogabrielides reviewed (commented):
https://github.com/dashpay/dash/blob/7aa5dcf2421072227fe5dfa2a43c85781c0f6344/src/evo/deterministicmns.h#L226

I am confused.. why here we pass basic BLS ?

ogabrielides commented:
@UdjinM6 Please check: https://github.com/ogabrielides/dash/commit/387b82efc3656384fd313690afffc20fe500c6e9

UdjinM6 commented:
implemented similar changes, fixed a couple of issues in migration logic and squashed [all these fixes](https://github.com/dashpay/dash/compare/23770b525affa9644526a5b2f79762eabd09a0d5..e8b0c1bae958ba8bc71ea61b18f21ee229bb3b7b) into original commits

UdjinM6 commented:
superseded by #5403
